src/box/lua/schema.lua

```diff
@@ -3684,13 +3685,65 @@ box.schema.user.passwd = function(name, new_password)
     end
 end
 
+-- Return origins for user/role from the given tuple.
+local function user_origins_from_tuple(tuple)
+    if tuple == nil then
+        return {[DEFAULT_ORIGIN] = true}
+    end
+    if tuple[USER_OPTS_FIELD_ID] == nil or
+       tuple[USER_OPTS_FIELD_ID].origins == nil then
+        return {[DEFAULT_ORIGIN] = true}
+    end
+    local res = {}
+    for name, _ in pairs(tuple[USER_OPTS_FIELD_ID].origins) do
+        res[name] = true
+    end
+    return res
+end
+
+-- Return opts field for user/role based on origins map.
+local function user_opts_from_origins(origins)
+    local normalized = {}
+    for name, v in pairs(origins) do
+        if v then
+            normalized[name] = true
+        end
+    end
```

Comment:
Nit: `v` can't be `nil` in the if-condition, because a table can't consist a `nil` value. It seems that the `normalized` table is always an equivalent to the input table `origins`.